activiti-services/activiti-services-connectors/src/main/java/org/activiti/services/connectors/behavior/MQServiceTaskBehavior.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.services.connectors.behavior;
+
+import java.util.UUID;
+
+import org.activiti.engine.delegate.DelegateExecution;
+import org.activiti.engine.impl.bpmn.behavior.AbstractBpmnActivityBehavior;
+import org.activiti.engine.impl.delegate.TriggerableActivityBehavior;
+import org.activiti.services.connectors.channel.ServiceTaskChannels;
+import org.activiti.services.connectors.model.AsyncContext;
+import org.activiti.services.connectors.model.ServiceTaskEvent;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MQServiceTaskBehavior extends AbstractBpmnActivityBehavior implements TriggerableActivityBehavior {
+
+    private final ServiceTaskChannels channels;
+
+    @Autowired
+    public MQServiceTaskBehavior(ServiceTaskChannels channels) {
+        this.channels = channels;
+    }
+
+    @Override
+    public void execute(DelegateExecution execution) {
+        AsyncContext context = new AsyncContext(execution.getProcessInstanceId(),
```

Comment:
@erdemedeiros look at the other comment with the steps to store these contexts in the DB. Here we should create the correlation ID so we don't send the executionId into the IntegrationEvent. 
It sounds like we will need a new engine internal service that will store this info to the DB.

Reply:
Ok. I'll work on that.